csr/csr.go

```diff
@@ -268,6 +272,101 @@ func getNodeCSRV1Beta1(ctx context.Context, client kubernetes.Interface, nodeNam
 	return nil, ErrNodeCertificateNotFound
 }
 
+func WatchAndApproveNodeCSRV1(ctx context.Context, log logrus.FieldLogger, client kubernetes.Interface, c chan *Certificate) error {
+	options := metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{
+			"spec.signerName": certv1.KubeAPIServerClientKubeletSignerName,
+		}).String(),
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			castai.LabelManagedBy: castai.LabelValueManagedByCASTAI,
+		}).String(),
+	}
+
+	w, err := client.CertificatesV1().CertificateSigningRequests().Watch(ctx, options)
+	if err != nil {
+		return err
+	}
+	defer w.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case event, ok := <-w.ResultChan():
+			if !ok {
+				return nil
+			}
+			csr, ok := event.Object.(*certv1.CertificateSigningRequest)
+			if !ok {
+				continue
+			}
+			ok, err := isNodeCSR(csr.Name, csr.Spec.Request, "")
+			if err != nil {
+				log.WithField("csr", csr.Name).Debugf("WatchAndApproveNodeCSRV1: skipping csr: %v", err)
+				continue
+			}
+			if !ok {
+				continue
+			}
+			sendCertificate(ctx, c, &Certificate{V1: csr})
+		}
+	}
+}
+func WatchAndApproveNodeCSRV1Beta1(ctx context.Context, log logrus.FieldLogger, client kubernetes.Interface, c chan *Certificate) error {
+	fs := fields.SelectorFromSet(fields.Set{
+		"spec.signerName": certv1beta1.KubeAPIServerClientKubeletSignerName,
+	})
+	ls := labels.SelectorFromSet(map[string]string{
+		castai.LabelManagedBy: castai.LabelValueManagedByCASTAI,
```

Comment:
I didn't see this selector on CSR.  Are you sure it is present?

Reply:
yeah, good point, I will remove it